uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsenceType.php

```diff
@@ -212,6 +212,15 @@ public static function getValuesArray($id) {
     return $absenceType;
   }
 
+  /**
+   * Returns a list of all possible categories for absence type
+   *
+   * @return array|bool
+   */
+  public static function getCategories() {
+    return array_flip(self::buildOptions('category'));
```

Comment:
Do we really need to flip this? Without flipping it returns an array of category labels indexed by the ID, which seems a more natural data set we would want this function to return.

Reply:
Also we need to add a test for this function.

Reply:
> Do we really need to flip this? Without flipping it returns an array of category labels indexed by the ID, which seems a more natural data set we would want this function to return.

That works well with label mapping implementation but could as well be re-written as
```
if (array_key_exists($categoryId, $this->categories)) {
  return $this->categories[$categoryId];
}
```
Or what do you think 🤷‍♂️